### ISSUE TITLE
Generating Entropy Files

### ISSUE THREAD

**Issue by Zuang99** [nabk89/NAS-with-Proxy-data#1]
Hello authors, I am really interested in your research, and I am trying to understand your research deeper. In this repo, you directly provided entropy files for 4 different datasets.

I am curious about the process of generting the entropy files. I tried using a trained model to evaluate the loss of every sample with cross entropy loss functoin, but the losses between samples are not as distinct as the files you provided. Would you like to post the code of generating the entropy list? I will really appreciate it!

**Comment by nabk89** [comment]
Thank you for your interest. I hope this answer helps you.

First, we used a github code (https://github.com/akamaster/pytorch_resnet_cifar10).
For CIFAR-10, ResNet-20 pretrained model in this repos was used, as mentioned in the paper.
For ImageNet, ResNet-50 pretrained model, which is available at PyTorch model (https://pytorch.org/vision/stable/models.html) was used.

Here is an example of the code to obtain the entropy values of all samples from the logits of them:
<pre><code>def entropy(data):
    tmp = F.softmax(data, dim=1) * F.log_softmax(data, dim=1)
    tmp = -1.0 * tmp.sum(dim=1)
    return tmp

entropy_list = [] # result
for i, (input, target) in enumerate(val_loader):
    input = input.cuda()
    output = model(input)        # model must be declared
    ent = entropy(output.data)   # entropy extraction
    entropy_list.extend(ent.tolist()) 
</code></pre>
